src/bundle/Resources/public/js/CKEditor/custom-attributes/ui/custom-attributes-form-view.js

```diff
@@ -209,8 +209,25 @@ class IbexaCustomAttributesFormView extends View {
         addListToDropdown(labeledDropdown.fieldView, itemsList);
 
         this.listenTo(labeledDropdown.fieldView, 'execute', (event) => {
-            const alreadySelectedValues = labeledDropdown.fieldView.element.value ? labeledDropdown.fieldView.element.value.split(' ') : [];
-            const value = config.multiple ? [...new Set([...alreadySelectedValues, event.source.value])].join(' ') : event.source.value;
+            const previousValue = labeledDropdown.fieldView.element.value;
+            let value;
+
+            if (config.multiple) {
+                value = previousValue ? [...new Set([...previousValue.split(' '), event.source.value])].join(' ') : event.source.value;
+            } else {
+                ({ value } = event.source);
+            }
+
+            if (previousValue === value) {
+                if (config.multiple) {
+                    const set = new Set([...previousValue.split(' ')]);
+
+                    set.delete(event.source.value);
+                    value = [...set].join(' ');
+                } else {
+                    value = null;
+                }
+            }
```

Comment:
I think we should try to simplify this.
We could do:
```js
const value = this.getNewValue(event.source.value, labeledDropdown.fieldView.element.value);
```
and introduce new method:
```js
    getNewValue(clickedValue, previousValue = '') {
        const selectedItems = new Set(previousValue.split(' '));

        if (selectedItems.has(clickedValue)) {
            selectedItems.delete(clickedValue);
        } else {
            selectedItems.add(clickedValue);
        }

        return [...selectedItems].join(' ');
    }
```

I'm not sure if the value must be null when empty because in single it is but in multiple it will be an empty string, if so the code should be adjusted.

Reply:
@dew326 I've tested it and in case of multiple empty string is present when the last value is removed. So it looks like it's working properly.